### PR TITLE
InviwoApp: Copy qt dll and plugins to bin folder

### PR DESCRIPTION
**Pull request by r-englund**
See #691

### CODE REVIEW

File: cmake/globalutils.cmake [hunk 1]
```diff
@@ -953,3 +953,18 @@ function(ivw_copy_if retval)
     set(${retval} ${res} PARENT_SCOPE)
 endfunction()
 
+function(ivw_deploy_qt target)
+if(WIN32)
```
Comment (petersteneteg) [inline cmake/globalutils.cmake]: use indentation

File: cmake/globalutils.cmake [hunk 2]
```diff
@@ -953,3 +953,18 @@ function(ivw_copy_if retval)
...
+find_program(WINDEPLOYQT_EXECUTABLE NAMES windeployqt HINTS ${QTDIR} ENV QTDIR PATH_SUFFIXES bin)
+
+get_filename_component(qt_bin_dir ${WINDEPLOYQT_EXECUTABLE} DIRECTORY  )
+add_custom_command(TARGET ${target} POST_BUILD 
+    COMMAND echo %PATH%
+    COMMAND qmake --version
+    COMMAND ${qt_bin_dir}/qtenv2.bat
+    COMMAND echo %PATH%
+    COMMAND qmake --version
```
Comment (petersteneteg) [inline cmake/globalutils.cmake]: is this just for debug?

File: cmake/globalutils.cmake [hunk 3]
```diff
@@ -953,3 +953,18 @@ function(ivw_copy_if retval)
     set(${retval} ${res} PARENT_SCOPE)
 endfunction()
 
+function(ivw_deploy_qt target)
+if(WIN32)
+find_program(WINDEPLOYQT_EXECUTABLE NAMES windeployqt HINTS ${QTDIR} ENV QTDIR PATH_SUFFIXES bin)
+
+get_filename_component(qt_bin_dir ${WINDEPLOYQT_EXECUTABLE} DIRECTORY  )
+add_custom_command(TARGET ${target} POST_BUILD
```
Comment (petersteneteg) [inline cmake/globalutils.cmake]: how many times will this run?

Comment (r-englund) [reply cmake/globalutils.cmake]: each time the target is built, its takes maybe a second to run

File: cmake/globalutils.cmake [hunk 4]
```diff
@@ -953,3 +953,14 @@ function(ivw_copy_if retval)
     set(${retval} ${res} PARENT_SCOPE)
 endfunction()
 
+function(ivw_deploy_qt target)
+    if(WIN32)
+        find_program(WINDEPLOYQT_EXECUTABLE NAMES windeployqt HINTS ${QTDIR} ENV QTDIR PATH_SUFFIXES bin)
+
+        get_filename_component(qt_bin_dir ${WINDEPLOYQT_EXECUTABLE} DIRECTORY  )
+        add_custom_command(TARGET ${target} POST_BUILD
```
Comment (ResearchDaniel) [inline cmake/globalutils.cmake:965]: Can't this be executed when running cmake instead?

File: cmake/globalutils.cmake [hunk 5]
```diff
@@ -953,3 +953,14 @@ function(ivw_copy_if retval)
     set(${retval} ${res} PARENT_SCOPE)
 endfunction()
 
+function(ivw_deploy_qt target)
```
Comment (ResearchDaniel) [inline cmake/globalutils.cmake:960]: Good to have a comment here explaining what it does

File: cmake/globalutils.cmake (hunk 4 above)
Comment (r-englund) [reply cmake/globalutils.cmake:965]: No, the `WINDEPLOYQT_EXECUTABLE` reads the built exe/dll to know what to copy, hence it needs to be `POST_BUILD`

**Comment by r-englund** [comment]
Anyone who wants to review?

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**